src/common/containers/builds.js

```diff
@@ -89,29 +55,15 @@ Builds.propTypes = {
     fullName: PropTypes.string.isRequired,
     url: PropTypes.string.isRequired
   }).isRequired,
-  isFetching: PropTypes.bool,
-  snap: PropTypes.shape({
-    self_link: PropTypes.string.isRequired,
-    store_name: PropTypes.string.isRequired
-  }),
-  success: PropTypes.bool,
-  error: PropTypes.object,
-  dispatch: PropTypes.func.isRequired
-};
-
-const mapStateToProps = (state, ownProps) => {
-  const owner = ownProps.params.owner.toLowerCase();
-  const name = ownProps.params.name.toLowerCase();
-  const fullName = `${owner}/${name}`;
-  const repository = state.repository;
-  // get builds for given repo from the store or set default empty values
-  const repoBuilds = state.snapBuilds[fullName] || snapBuildsInitialStatus;
-
-  return {
-    fullName,
-    repository,
-    ...repoBuilds
-  };
+  snapBuilds: PropTypes.shape({
+    isFetching: PropTypes.bool,
+    snap: PropTypes.shape({
+      self_link: PropTypes.string.isRequired,
+      store_name: PropTypes.string.isRequired
+    }),
+    success: PropTypes.bool,
+    error: PropTypes.object
+  })
 };
 
-export default connect(mapStateToProps)(withRepository(Builds));
+export default withRepository(withSnapBuilds(Builds));
```

Comment:
I think this is a good trade off. Wrapping / container components like this have a specific role: grouping presentational components together to make a user interface. I do not think we will want to re-use these much.